charms/katib-db-manager/src/charm.py

```diff
@@ -4,123 +4,137 @@
-            self.model.unit.status = check_failed.status
-            return
-
-        self.model.unit.status = MaintenanceStatus("Setting pod spec")
-
-        self.model.pod.set_spec(
-            {
-                "version": 3,
-                "serviceAccount": {
-                    "roles": [
-                        {
-                            "global": True,
-                            "rules": [
-                                {
-                                    "apiGroups": [""],
-                                    "resources": ["configmaps", "namespaces"],
-                                    "verbs": ["*"],
-                                },
-                                {
-                                    "apiGroups": ["kubeflow.org"],
-                                    "resources": [
-                                        "experiments",
-                                        "trials",
-                                        "suggestions",
-                                    ],
-                                    "verbs": ["*"],
-                                },
-                            ],
-                        }
-                    ]
-                },
-                "containers": [
-                    {
-                        "name": "katib-db-manager",
-                        "command": ["./katib-db-manager"],
-                        "imageDetails": image_details,
-                        "ports": [{"name": "api", "containerPort": self.model.config["port"]}],
-                        "envConfig": {
-                            "DB_NAME": "mysql",
-                            "DB_USER": "root",
-                            "DB_PASSWORD": mysql_data["root_password"],
-                            "KATIB_MYSQL_DB_HOST": mysql_data["host"],
-                            "KATIB_MYSQL_DB_PORT": mysql_data["port"],
-                            "KATIB_MYSQL_DB_DATABASE": mysql_data["database"],
-                        },
-                        "kubernetes": {
-                            "livenessProbe": {
-                                "exec": {
-                                    "command": [
-                                        "/bin/grpc_health_probe",
-                                        f"-addr=:{self.model.config['port']}",
-                                    ]
-                                },
-                                "initialDelaySeconds": 10,
-                                "periodSeconds": 60,
-                                "failureThreshold": 5,
-                            }
-                        },
-                    }
-                ],
-            }
+        # retrieve configuration and base settings
+        self.logger = logging.getLogger(__name__)
+        self._container_name = "katib-db-manager"
+        self._container = self.unit.get_container(self._container_name)
+        self._exec_command = "./katib-db-manager"
```

Comment:
is there a reason to put exec command here and not directly in the layer?
do we use it anywhere else?

Reply:
We've been doing this way for all of our charms. No particular reason, just keeping all container related config together: container name, container exec command.

Reply:
general rule in coding keep your variables close to the executable code.

this comment is not blocking the PR, but please reconsider the pattern